### PR TITLE
Add Canary (NeMo Conformer-AED) ONNX STT

### PR DESCRIPTION
**Pull request by hagope**
## Summary

Adds `OnnxCanaryStt`, an `STTInterface` implementation for NVIDIA Canary — FastConformer encoder plus an autoregressive Transformer decoder — against the ONNX exports published for [onnx-asr](https://github.com/istupakov/onnx-asr) (`istupakov/canary-*-onnx`).

Offline per utterance: the encoder consumes the whole segment, then tokens are decoded one at a time. No streaming mode, no partial results, and contextual biasing does not apply. That suits the VAD-segmented pipeline, which already hands STT a complete utterance.

## What it reuses

The front end is Parakeet's, unchanged: pre-emphasis, 128-bin log-mel at 512/160/400, then NeMo per-feature normalisation. Both are FastConformer encoders trained by the same `AudioToMelSpectrogramPreprocessor`, and Parakeet is already at 128 bins, so nothing new was needed for features.

## Decoding

Greedy, with the `decoder_mems` cache:

```
encoder: audio_signal[1,128,T], length[1] -> encoder_embeddings, encoder_mask
decoder: input_ids, encoder_embeddings, encoder_mask, decoder_mems
         -> logits, decoder_hidden_states
```

The ten-token control prompt goes in on the first step; afterwards only the newest token, with the returned hidden states fed back as the cache. `decoder_mems` layer count and width are read from the export rather than hard-coded, so the same wrapper serves 180m-flash, 1b-flash and 1b-v2.

Source and target language are separate prompt tokens, so translation is reachable from the same model later; only transcription is wired up here.

## A note on the vocab

The published vocab has **no bare-space token** — the space is SentencePiece's U+2581. onnx-asr looks up `" "` because its loader substitutes the marker at load time; a port that reads the file literally gets `-1`.

That is worth guarding rather than commenting, because the failure is not obvious: a `-1` in the prompt makes the decoder emit fluent text that either stops after two words or repeats a fragment until the token cap. It reads as a KV-cache bug, and I chased it as one before finding the real cause. Missing prompt tokens now throw at construction.

## Measurements

Galaxy S26 (Snapdragon, arm64), INT8, three Kokoro-synthesized utterances, identical audio across models, via speech-android:

| | Parakeet TDT | Whisper Small | **Canary 180M** |
| --- | --- | --- | --- |
| decode, mean | 451 ms | 1,485 ms | **450 ms** |
| load | 5,126 ms | 4,482 ms | **3,449 ms** |
| native heap held | 1,452 MB | 857 MB | **798 MB** |
| download | 891 MB | 374 MB | **213 MB** |

Transcripts were comparable; Canary's casing and punctuation were the cleanest of the three. All three missed the same proper noun.

Three synthesized phrases is a smoke test, not a WER benchmark — TTS audio is cleaner than real speech. Directional only.

## Test plan

- `cmake -B build -DCMAKE_BUILD_TYPE=Release && cmake --build build && ctest` — 22/22 pass
- `SPEECH_CORE_WITH_ONNX=ON` verified through speech-android's Android build (arm64-v8a and x86_64)
- On-device transcription through the full pipeline on a Galaxy S26, numbers above
- No unit test for the wrapper itself: it needs the 213 MB model bundle, which the existing model wrappers also lack coverage for

## Notes

`CMakeLists.txt` gains one source inside the existing `SPEECH_CORE_WITH_ONNX` block; the default configuration is unchanged.

The speech-android side (SttModel entry, JNI branch, ModelManager files) is ready and waits on this landing, per the merge sequence in the repo guidelines.

### CODE REVIEW

**Comment by ivan-digital** [comment]
Re-exported Canary ourselves. Bundle live at [soniqo/Canary-180M-Flash-ONNX](https://huggingface.co/soniqo/Canary-180M-Flash-ONNX), wrapper targets it; Core ML export at [aufklarer/Canary-180M-Flash-CoreML](https://huggingface.co/aufklarer/Canary-180M-Flash-CoreML).

The bundle carries its decode contract — prompt ids, cache dimensions, end-of-text — in `config.json` and graph metadata, so the wrapper does no token-string lookups and no shape probing.

All numbers below: FLEURS, 100 utterances per language (Spanish slice had 11), INT8, greedy, ONNX Runtime on CPU with 4 threads, Apple silicon.

**Prompt** — was ten tokens, NeMo's formatter returns nine:

| WER | en | de | es | fr |
|---|---|---|---|---|
| nine tokens | 9.24 | 7.23 | 1.97 | 10.43 |
| with leading `▁` | 13.96 | 13.56 | 6.58 | 15.20 |
| `istupakov` export, same change | 9.15 → 15.79 | 6.93 → 15.48 | 3.29 → 10.53 | 10.74 → 13.73 |

**Front end** — mel defaults vs the contract Parakeet follows for the same preprocessor:

| WER | en | de | es | fr |
|---|---|---|---|---|
| trained contract | 9.24 | 7.23 | 1.97 | 10.43 |
| mel defaults | 9.57 | 7.87 | 1.97 | 12.45 |

**Ours vs the export your numbers came from**, same machine, 20 English utterances for the runtime figures, measured in a process holding only ONNX Runtime:

| | ours | reference |
|---|---|---|
| WER en / de / es / fr | 9.24 / 7.23 / 1.97 / 10.43 | 9.15 / 6.93 / 3.29 / 10.74 |
| latency | 186 ms | 211 ms |
| RTF | 0.0198 | 0.0223 |
| throughput | 50.6× | 44.7× real time |
| peak RSS | 920 MB | 964 MB |
| on load | +584 MB | +662 MB |
| on disk | 273 MB | 213 MB |

The 60 MB is their quantized convolutions (+40) and embedding table (+20); ours leaves both FP32.

**Quantization**, ours, identical transcripts across all three:

| | disk | peak RSS | latency | throughput |
|---|---|---|---|---|
| INT8 | 273 MB | 920 MB | 186 ms | 50.6× |
| INT8 enc + FP32 dec | 490 MB | 1136 MB | 253 ms | 37.2× |
| FP32 | 778 MB | 1395 MB | 252 ms | 37.4× |

Also fixed: neither ORT session was released, OrtValues leaked on error paths, non-16 kHz returned an empty transcript instead of resampling, no `cancel()`.

Added `docs/models.md`, README rows across the mirrors, `scripts/download_canary_onnx.sh`, and a bundle-gated test. Merged with main, 39/39 passing.

S26 table predates this bundle — worth re-running on device.
